File: content-solutions/queue-deque/queue-front-back.py
from collections import deque


def solution(commands):
    line = deque()
    answer = []
    for command in commands:
        parts = command.split()
        name = parts[0]
        if name == "enqueue":
            line.append(int(parts[1]))
        elif name == "dequeue":
            if line:
                line.popleft()
        elif name == "front":
            answer.append(line[0] if line else -1)
        else:
            answer.append(line[-1] if line else -1)
    return answer

File: content-solutions/queue-deque/queue-front-back.js
function solution(commands) {
  const line = [];
  let head = 0;
  const answer = [];
  for (const command of commands) {
    const [name, value] = command.split(" ");
    if (name === "enqueue") line.push(Number(value));
    else if (name === "dequeue") {
      if (head < line.length) head += 1;
    } else if (name === "front") answer.push(head < line.length ? line[head] : -1);
    else answer.push(head < line.length ? line[line.length - 1] : -1);
  }
  return answer;
}
